File: 14주차/이연상/빗물.py
# [BOJ] 14719번_빗물

h, w = map(int, input().split())
nums = list(map(int, input().split()))

graph = [ [0 for j in range(500)] for i in range(500)]
for i in range(len(nums)):
    for j in range(nums[i]):
        graph[j][i] = 1

res = 0
for y in range(500):
    temp = False
    val = 0
    for x in range(500):
        
        if graph[y][x] == 1: # 1 만났을 때
            if temp == True:
                if val != 0:
                    res += val
                    val = 0
            else:
                temp = True

        else:  # 0 만났을 때
            if temp == True:
                val += 1

print(res)

        







                

File: 54주차/이연상/빗물.java
// [BOJ] 빗물

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int h = sc.nextInt();
        int w = sc.nextInt();
        int[] nums = new int[w];

        for (int i = 0; i < w; i++) {
            nums[i] = sc.nextInt();
        }

        int[][] graph = new int[500][500];
        for (int i = 0; i < nums.length; i++) {
            for (int j = 0; j < nums[i]; j++) {
                graph[j][i] = 1;
            }
        }

        int res = 0;
        for (int y = 0; y < 500; y++) {
            boolean temp = false;
            int val = 0;
            for (int x = 0; x < 500; x++) {

                if (graph[y][x] == 1) { // 1 만났을 때
                    if (temp) {
                        if (val != 0) {
                            res += val;
                            val = 0;
                        }
                    } else {
                        temp = true;
                    }
                } else { // 0 만났을 때
                    if (temp) {
                        val++;
                    }
                }
            }
        }

        System.out.println(res);
        sc.close();
    }
}